plutus-tx/src/PlutusTx/Utils.hs

```diff
@@ -7,6 +7,4 @@ mustBeReplaced :: String -> a
 mustBeReplaced placeholder =
   error $
     "The " <> show placeholder <> " placeholder must have been replaced by the \
-      \core-to-plc plugin during compilation. \
-      \The most likely reason the replacement didn't happen is that a PlutusTx \
-      \function was evaluated by off-chain code."
+      \core-to-plc plugin during compilation."
```

Comment:
When I added this message:
_"The most likely reason the replacement didn't happen is that a PlutusTx function was evaluated by off-chain code."_ my motivation was to help users understand **why** the error happens, as in what causes it.

Otherwise, this message alone "_The 'foo' placeholder must have been replaced by the core-to-plc plugin during compilation._" could be frustrating as it doesn't help them to understand and  fix the cause.

Reply:
My understanding is that "placeholder" refers to `PlutusTx.Plugin.Utils.plc`. The plugin looks for `plc`, and replaces it with the actual compilation.

If the replacement does not happen, it is not because "a PlutusTx function was evaluated by off-chain code" (which one _can_ do - I don't think there's anything wrong with that). It is either because one uses `plc` directly but forgets to use the plugin, a bug in the plugin, or `plc` is used in a way that is not recognized by the plugin (the plugin requires `plc` to be applied to the right arguments to be triggered).

Reply:
Is seems to me that "a PlutusTx function was evaluated by off-chain code" happens when "one uses plc directly but forgets to use the plugin", right?

In either case, you list of reasons could be really useful for users to understand what they did wrong (modulo bug in the plugin), I'd extend the original message, not reduce it.

Reply:
> Is seems to me that "a PlutusTx function was evaluated by off-chain code" happens when "one uses plc directly but forgets to use the plugin", right?

I don't think so. "forgetting to use the plugin" means forgetting `{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}`, which is totally different from evaluating Plutus Tx functions in off-chain code, which is a perfectly OK thing to do.